Lib/ufo2ft/featureWriters/kernFeatureWriter.py

```diff
@@ -350,7 +350,25 @@ def _makePairPosRule(self, pair, side1Classes, side2Classes, rtl=False):
     def _makeKerningLookup(self, name, ignoreMarks=True):
         lookup = ast.LookupBlock(name)
         if ignoreMarks and self.options.ignoreMarks:
-            lookup.statements.append(ast.makeLookupFlag("IgnoreMarks"))
+            # We only want to filter the spacing marks
+            marks = self.context.gdefClasses.mark
+            spacing = []
+            if marks:
+                spacing = [mark for mark in marks if self.context.font[mark].width != 0]
+            if not spacing:
```

Comment:
How does this work? Many font sources have none spacing marks with +ve advances and rely on the compiler to zero them, would they be considered spacing marks in this context? Or the zeroing would have happened already at this point (but this cide would be noop then)?

Reply:
If this depends on Glyphs’s “Spacing Combining” sub category, then it probably should have checked for this, checking for glyph width feels like the kind of indirection that comes back to bite you in the rear.

Reply:
Only non-spacing combining marks are zeroed by the compiler, spacing ones are not and as such have a non zero advance width.

Reply:
But spacing combining marks do not exist in OpenType nor AFDKO feature files, only in Glyphs, so what happens to sources not coming from Glyphs that happen to have marks with advance widths (which is harmless as the layout engine will be zerowing them for most scripts)?

Reply:
If your sources say that a glyph has advance width, you should not be surprised when ufo2ft treats it as a glyph with advance width.

(And remember, the change here only affects the very specific case of ignoring marks in kerning between two bases. It doesn't change anything that's already happening or not happening with regard to mark width zeroing.)

Reply:
> If your sources say that a glyph has advance width, you should not be surprised when ufo2ft treats it as a glyph with advance width.

They have the right glyph class, the fact they have sn advance width is irrelevant as the layout engine will zero them when shaping.

Reply:
> > spacing combining marks do not exist in OpenType nor AFDKO feature files, only in Glyphs
> 
> well, "spacing combining marks" do exist in unicode, general category "Mc", don't they?

But we are not using that here, and it is also irrelevant as OpenType knows only about only one class of mark glyphs and the font sets the glyph class, but we are second guessing its intentions based on a tangentially relevant heuristic. The whole concept of feature writers is broken beyond repair, but adding more surprising exceptions and heuristics tuned for individual fonts is not helping.

Reply:
Here is a more real example. Some monospace fonts, e.g. DejaVu Sans Mono, give combining marks an advance width (so that legacy systems that require all glyphs in monospace fonts to have the same adcance width would still recognize them as monospace) then zeros the width using GPOS single positioning lookups. Monospace fonts obviously don't need kerning, but this shows a combining mark can have an advance width while not being a spacing combining one.